lib/vnet/network_stack.go

```diff
@@ -274,6 +280,13 @@ func createStack() (*stack.Stack, *channel.Endpoint, error) {
 	if err := netStack.CreateNIC(nicID, linkEndpoint); err != nil {
 		return nil, nil, trace.Errorf("creating VNet NIC: %s", err)
 	}
+	// Put the NIC into promiscuous mode so that VNet can handle incoming IP packets
+	// on addresses that have not been explicitly assigned to any handler yet.
+	// This is so the DNS server can handle UDP requests to port 53 on any IP
+	// address.
+	if err := netStack.SetPromiscuousMode(nicID, true); err != nil {
```

Comment:
Are there any other implications to this change?

Reply:
Not much, this does not effect the TUN or the host in any way, it only changes the mode on the gVisor NIC. The effect is that when gVisor receives an incoming IP packet from the host where the destination IP does not match an IP that has been explicitly assigned to the gVisor NIC, it would still call VNet's TCP or UDP handler instead of just dropping the packet. But it looks like it does assign some internal state for each unique IP used in this way so there could be some memory overhead if a client hit a ton of IPs in VNet's range.

This was always kind of a lazy way of doing this, in the latest commit i switched to explicitly assigning the address for the IPv4 DNS server and avoiding promiscuous mode.